src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,107 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useState, useEffect, useCallback } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface TodoFilterProps {
+  todos: Todo[];
+  handleFilterChange: (filteredTodos: Todo[]) => void;
+}
+
+export enum FilterOptions {
+  ALL = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  todos,
+  handleFilterChange,
+}) => {
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.ALL);
+  const [query, setQuery] = useState('');
+
+  const handleFilterTodos = () => {
+    let filteredTodos = todos.filter(todo => {
+      switch (filter) {
+        case FilterOptions.ACTIVE:
+          return !todo.completed;
+        case FilterOptions.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+
+    if (query) {
+      filteredTodos = filteredTodos.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    handleFilterChange(filteredTodos);
+  };
```

Comment:
You're interacting with DOM directly. In React, we should avoid direct DOM manipulation as much as possible and let React handle the DOM updates. Instead of filtering todos in the component, it would be better to pass the filter value and query to the parent component and let it handle the filtering.